### ISSUE TITLE
cat: {my prefix}.callsites*.3.txt: No such file or directory

### ISSUE THREAD

**Issue by JieWu2012** [liucongcas/GLORI-tools#23]
Hi， 

When I run_GLORi.py, I got an error like the title at the end of the running. I got the "totalCR.txt" and "totalformat.txt". But no "csv" file. Can you help me out why there are no "callsites*.3" generated?

**Comment by yanxueqing621** [comment]
hi, I also occurred this problem，which is due to low conversion rate caused by conversion failure during library construction.  I thank that conversion failure result in no m6a site identified, so no "callsites*.3" file will be created. 

Hope it will help.

**Comment by wangyuhu123** [comment]
Hey, I have the same problem as you. Did you solve it?

**Comment by JianhuaWang95** [comment]
Hey, any update for this?

**Comment by vetmohit89** [comment]
I am also having similar issue. @liucongcas can you please respond?

**Comment by joshtburdick** [comment]
When I initially tried running this, I neglected to trim adapters. (This was possibly a silly mistake.) This meant that abour 3% of reads were aligning, and unsurprisingly, no sites passed the thresholds.

After trimming adapters, mapping rate was around 80%, `"callsites*.3"` files were generated, and some sites passed the thresholds.

This may or may not be the issue you're running into, or related.

I removed duplicate reads using `fastp`, since `seqkit rmdup` seemed to be using a lot of memory (similarly to what others have noted). Admittedly, I wasn't using the latest version of `seqkit`.

**Comment by vetmohit89** [comment]
I have generated merged fastq file using fastap command (may be this could be one of the reason). Can you please point out how exactly you have prepared your input fastq file

**Comment by joshtburdick** [comment]
To trim reads, I used the steps in the GLORI-seq README, except that instead of `seqkit rmdup`, I used `fastp` as follows.
```
fastp \
-i ${output_dir}/${output_base}_trimmed.fq \
-o ${output_dir}/${output_base}.rmdup.fq \
--json ${output_dir}/${output_base}.fastp.json \
--thread 8 --dedup --dup_calc_accuracy 6 \
--disable_adapter_trimming --disable_trim_poly_g --disable_quality_filtering --disable_length_filtering
```

You'll need to set variables such as `output_dir` and `output_base1`, or hard-code them for testing.

I was trying to use `fastp` only for duplicate removal. It's possible that `fastp` could do some of the other steps as well. But in the interests of following the README as closely as possible, I didn't try to do that...

**Comment by BrendanBeahan** [comment]
@joshtburdick Sorry to ask, but were you able to reproduce the initial glori paper's results using these changes? I have been trying to run glori-seq on their public [data](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM6432590) but I keep failing to get anywhere near the 214k sites they cite in their paper.

**Comment by joshtburdick** [comment]
I've only run it on the HEK293 samples. With that, it printed a comparable number of sites (212k and 217k). Of those, for each replicate, about 166k of the sites overlapped with the 170k in the table for HEK293. The correlation of m6A level (Pearson r) was > 0.998. So I'd say it reproduced those.

Note that this is using the FASTQ trimming by `fastp` as described above, and after folding in several changes mentioned here on Github. I could post those (and potentially file a pull request, although I don't really know the procedure for that). I could possibly also post the Slurm scripts I used, but your Nextflow workflow is apt to be cleaner...

**Comment by BrendanBeahan** [comment]
@joshtburdick Indeed, would it be at all possible to post your final slurm scripts? I would be very appreciative and I think it would help quite a bit in determining what I am doing wrong. I set up my NF script and Dockerfile as far as I can tell matching the recomendations of the authors and with some of the changes mentioned in other github issues, but I am still not getting site overlap or a comparable # of sites on the HEK datasets. 

If I can determine what is wrong, perhaps I can also publicly surface my NF pipeline and Dockerfile to make things easier for others going forward.

**Comment by joshtburdick** [comment]
I have forked this, at [https://github.com/joshtburdick/GLORI-tools-1](https://github.com/joshtburdick/GLORI-tools-1). I've tried to include the updates I made (most of which were from this forum).

There's no sense in which I've tested the code after pulling from that repo, and it will need some configuration anyway. If you run into issues with it, if you post an issue on the fork, I'll try to look into it. (Or here, if it's an upstream issue.)

I'll be curious to hear if this helps figure out what issue you're running into.

**Comment by BrendanBeahan** [comment]
@joshtburdick Thank you so much, Josh—that was really kind of you. I’ll pull your fork and try it out. I’ll let you know whether it resolves the problem or if I run into anything else. Thanks again!

**Comment by BrendanBeahan** [comment]
@joshtburdick Well your scripts certainly helped. For some silly reason I had set my UMI variable to 5 instead of the requisite 10... my NF and Dockerfile appear to work properly and recpature a similarly high amount of the original output m6A hits with almost total overlap now. 

I suppose I will set up the container and other files for wider use shortly.  Thanks again!

**Comment by Chi7hang** [comment]
It is caused by a minor issue in line 120 of m6A_caller.py. The indentation was 2 tabs before try:. This issue will raise TabError: inconsistent use of tabs and spaces in indentation, and terminate without return, so no {}.callsites.chrN.N.txt will be written. Replacing 2 tabs with 16 of spaces will fix the issue.

Fewer aligned reads was not a problem for me. I sequenced only 13M reads for a test, then it yielded 4.3M alignments. After fixing the issue, ~10K sites were called.

**Comment by Chi7hang** [comment]
> It is caused by a minor issue in line 120 of m6A_caller.py. The indentation was 2 tabs before try:. This issue will raise TabError: inconsistent use of tabs and spaces in indentation, and terminate without return, so no {}.callsites.chrN.N.txt will be written. Replacing 2 tabs with 16 of spaces will fix the issue.
> 
> Fewer aligned reads was not a problem for me. I sequenced only 13M reads for a test, then it yielded 4.3M alignments. After fixing the issue, ~100K sites were called.

Well, it was already patched by author on May 18, 2024.
